spring-pulsar/src/main/java/org/springframework/pulsar/core/DefaultPulsarConsumerFactory.java

```diff
@@ -77,10 +79,19 @@ public Consumer<T> createConsumer(Schema<T> schema, BatchReceivePolicy batchRece
 		final Map<String, Object> properties = new HashMap<>(this.consumerConfig);
 		properties.putAll(propertiesToOverride);
 
+		DeadLetterPolicy deadLetterPolicy = null;
+		if (properties.containsKey("deadLetterPolicy")) {
+			deadLetterPolicy = (DeadLetterPolicy) properties.remove("deadLetterPolicy"); // https://github.com/apache/pulsar/issues/11646
+		}
+
 		if (!CollectionUtils.isEmpty(properties)) {
 			consumerBuilder.loadConf(properties);
 		}
 
+		if (deadLetterPolicy != null) {
+			consumerBuilder.deadLetterPolicy(deadLetterPolicy);
+		}
+
```

Comment:
Here we have to use this ugly workaround because `loadConf` cannot read in the DeadLetterPolicy correctly

Reply:
Thanks for adding the comment in the code as well @alpreu